datasource/src/main/scala/quasar/physical/mongo/Tunnel.scala

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright 2014–2018 SlamData Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package quasar.physical.mongo
+
+import slamdata.Predef._
+
+import cats.effect._
+import cats.syntax.applicative._
+import cats.syntax.functor._
+import cats.syntax.flatMap._
+import cats.syntax.apply._
+
+import com.jcraft.jsch._
+import com.mongodb.connection.ClusterSettings
+import com.mongodb.{ConnectionString, ServerAddress}
+
+import quasar.Disposable
+
+import scala.collection.JavaConverters._
+
+object Tunnel {
+  import TunnelConfig._, Pass._
+
+  val SessionName: String = "default"
+
+  def apply[F[_]: Sync: ContextShift](config: MongoConfig): F[Disposable[F, ClusterSettings]] =
```

Comment:
Do we need to use `Disposable` here? Or can we use `Stream` or `Resource`? `Disposable` is technically unsafe as the scoping is unconstrained, so we should only use it for resources which have non-lexical lifecycles.

Reply:
The tunnel should be closed when mongo-ds instance is disposed